apps/site/lib/site_web/controllers/helpers.ex

```diff
@@ -76,10 +76,11 @@ defmodule SiteWeb.ControllerHelpers do
         opts
       ) do
     filter_by_direction? = Keyword.get(opts, :filter_by_direction?, true)
+    repo_fn = Keyword.get(opts, :repo_fn, &Repo.by_route_id_and_type/3)
 
     alerts =
       route_id
-      |> Repo.by_route_id_and_type(route_type, date_time)
+      |> repo_fn.(route_type, date_time)
```

Comment:
The name `repo_fn` seems like it could be more descriptive here, but other than going full `alerts_by_route_id_and_type_fn` I've got nothing. It would be nice if we had a more structured means of doing this kind of "dependency injection" in the first place, but in the short term I do appreciate the testing enabled by doing it here.

Reply:
@digitalcora let's go with `alerts_by_route_id_and_type_fn` then, that's a perfectly cromulent name.

Reply:
Ok, have fun with the line wrapping 😄